ledger/sandbox-classic/src/test/lib/scala/platform/sandbox/auth/MultiPartyServiceCallAuthTests.scala

```diff
@@ -0,0 +1,211 @@
+      serviceCallFor(
+        TokenParties(List(randomParty), List.empty),
+        RequestSubmitters(singleParty, List.empty, List.empty),
+      ))
+  }
+  it should "deny single-party calls authorized to a random party (actAs)" in {
+    expectPermissionDenied(
+      serviceCallFor(
+        TokenParties(List(randomParty), List.empty),
+        RequestSubmitters("", List(singleParty), List.empty),
+      ))
+  }
+
+  it should "allow multi-party calls authorized to exactly the required parties" in {
+    // Note: use expectSuccess() once multi-party submissions are enabled
+    expectUnimplemented(
+      serviceCallFor(
+        TokenParties(actAs, readAs),
+        RequestSubmitters("", actAs, readAs),
+      ))
+  }
+  it should "allow multi-party calls authorized to a superset of the required parties" in {
+    // Note: use expectSuccess() once multi-party submissions are enabled
+    expectUnimplemented(
+      serviceCallFor(
+        TokenParties(randomActAs ++ actAs, randomReadAs ++ readAs),
+        RequestSubmitters("", actAs, readAs),
+      ))
+  }
+  it should "allow multi-party calls with all parties authorized in read-write mode" in {
+    // Note: use expectSuccess() once multi-party submissions are enabled
+    expectUnimplemented(
+      serviceCallFor(
+        TokenParties(actAs ++ readAs, List.empty),
+        RequestSubmitters("", actAs, readAs),
+      ))
+  }
+  it should "allow multi-party calls with actAs parties spread across party and actAs fields" in {
+    // Note: use expectSuccess() once multi-party submissions are enabled
+    expectUnimplemented(
+      serviceCallFor(
+        TokenParties(actAs, readAs),
+        RequestSubmitters(actAs.head, actAs.tail, readAs),
+      ))
+  }
+  it should "allow multi-party calls with actAs parties duplicated in the readAs field" in {
+    // Note: use expectSuccess() once multi-party submissions are enabled
+    expectUnimplemented(
+      serviceCallFor(
+        TokenParties(actAs, readAs),
+        RequestSubmitters("", actAs, actAs ++ readAs),
+      ))
+  }
+
+  it should "deny multi-party calls authorized to no parties" in {
+    expectPermissionDenied(
+      serviceCallFor(
+        TokenParties(List.empty, List.empty),
+        RequestSubmitters("", actAs, readAs),
+      ))
+  }
+  it should "deny multi-party calls authorized to random parties" in {
+    expectPermissionDenied(
+      serviceCallFor(
+        TokenParties(randomActAs, randomReadAs),
+        RequestSubmitters("", actAs, readAs),
+      ))
+  }
+  it should "deny multi-party calls with all parties authorized in read-only mode" in {
+    expectPermissionDenied(
+      serviceCallFor(
+        TokenParties(List.empty, actAs ++ readAs),
+        RequestSubmitters("", actAs, readAs),
+      ))
+  }
+  it should "deny multi-party calls with one missing actor authorization" in {
+    expectPermissionDenied(
+      serviceCallFor(
+        TokenParties(actAs.take(actorsCount - 1), readAs),
```

Comment:
```suggestion
        TokenParties(actAs.drop(1), readAs),
```
or  `actAs.dropRight(1)` or `actAs.init` if you want to drop the last element.

Reply:
```scala
scala> List.empty.init
java.lang.UnsupportedOperationException: empty.init
  at scala.collection.TraversableLike.init(TraversableLike.scala:586)
  at scala.collection.TraversableLike.init$(TraversableLike.scala:585)
  at scala.collection.AbstractTraversable.init(Traversable.scala:108)
  ... 28 elided

scala> List.empty.dropRight(1)
res1: List[Nothing] = List()
```